scheduler/scheduler.go

```diff
@@ -156,10 +156,11 @@ func (fq *fifoQueue) wait(ctx context.Context) (func(), error) {
 	fq.mu.RUnlock()
 
 	// handle queue
-	all := cnt.Inc()
+	all := cnt.Add(1)
 	// queue full?
 	if all > maxConcurrency+maxQueueSize {
-		cnt.Dec()
+		// Decrement counter
+		cnt.Store(cnt.Load() - 1)
```

Comment:
Not atomic anymore. Internally it uses https://pkg.go.dev/sync/atomic#AddUint64 that suggests that `cnt.Add(^uint64(0))` would work.

I am not sure why we want uint here, likely we can just change to `atomic.Int64` and use `cnt.Add(-1)`

Reply:
true atomic.Int64 should be fine

Reply:
Updated to `atomic.Int64` in the new patchset. However, because of various arithmetic operations with `maxQueueSize` and `maxConcurrency`, had to change their type from `uint64` to `int64` as well, to make operations compatible.